File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] arr) {
        int count = 1;
        int maj = arr[0];
        for(int i = 1;i<arr.length;i++){
            if(count==0){
                maj = arr[i];
                count = 1;
            }
            else if(maj==arr[i]){
                count++;
            }
            else{
                count--;
            }
        }
        return maj;
    }
}

File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        for(int i=0; i<nums.size(); i++){
            int target = nums[i],count = 0;
            for(int j=i; j<nums.size(); j++){
                if(nums[j]==target){
                    count++;
                }

            }
            if(count>nums.size()/2){
            return  target;
            }
        }
        return -1;
        
    }
};